### ISSUE TITLE
`W292` missing detection

### ISSUE THREAD

**Issue by boolean-light** [astral-sh/ruff#10379]
## Summary:
If I create `test.py` with the content:
```python
print(2)
```
This should throw error of `W292 no newline at end of file`, but it doesn't.

## What I investigated:
Although there is `missing_newline_at_end_of_file.rs` in `crates/ruff_linter/src/rules/pycodestyle/rules`, it seems that the function `check_physical_lines` in `crates/ruff_linter/src/checkers/physical_lines.rs` is not called in `crates/ruff_linter/src/linter.rs`.
From line 210 to line 213, there is this `settings.rules.iter_enabled().any(|rule_code| rule_code.lint_source.is_physical_lines())`, but all of those returned false.
FYI if I inserted `settings.rules.iter_enabled().for_each(|rule_code| {println!("{:?}", rule_code)});`, I got this:
```
MultipleImportsOnOneLine
ModuleImportNotAtTopOfFile
MultipleStatementsOnOneLineColon
MultipleStatementsOnOneLineSemicolon
UselessSemicolon
NoneComparison
TrueFalseComparison
NotInTest
NotIsTest
TypeComparison
BareExcept
LambdaAssignment
AmbiguousVariableName
AmbiguousClassName
AmbiguousFunctionName
IOError
SyntaxError
UnusedImport
ImportShadowedByLoopVar
UndefinedLocalWithImportStar
LateFutureImport
UndefinedLocalWithImportStarUsage
UndefinedLocalWithNestedImportStarUsage
FutureFeatureNotDefined
PercentFormatInvalidFormat
PercentFormatExpectedMapping
PercentFormatExpectedSequence
PercentFormatExtraNamedArguments
PercentFormatMissingArgument
PercentFormatMixedPositionalAndNamed
PercentFormatPositionalCountMismatch
PercentFormatStarRequiresSequence
PercentFormatUnsupportedFormatCharacter
StringDotFormatInvalidFormat
StringDotFormatExtraNamedArguments
StringDotFormatExtraPositionalArguments
StringDotFormatMissingArguments
StringDotFormatMixingAutomatic
FStringMissingPlaceholders
MultiValueRepeatedKeyLiteral
MultiValueRepeatedKeyVariable
ExpressionsInStarAssignment
MultipleStarredExpressions
AssertTuple
IsLiteral
InvalidPrintSyntax
IfTuple
BreakOutsideLoop
ContinueOutsideLoop
YieldOutsideFunction
ReturnOutsideFunction
DefaultExceptNotLast
ForwardAnnotationSyntaxError
RedefinedWhileUnused
UndefinedName
UndefinedExport
UndefinedLocal
UnusedVariable
UnusedAnnotation
RaiseNotImplemented
```
which doesn't include `MissingNewlineAtEndOfFile`.
Sorry I wasn't very sure about the internal system of where all of these rules are defined, so I couldn't investigate more, but hope this information helps.

**Comment by dhruvmanila** [comment]
Thanks for investigating into it and providing all the details!

Can you provide some additional details which I can use to try reproducing the possible bug as it's giving me the diagnostic with `v0.3.2`:

```console
$ ruff check --select=W292 --isolated --no-cache --output-format=full src/W292.py
src/W292.py:1:9: W292 [*] No newline at end of file
  |
1 | print(2)
  |          W292
  |
  = help: Add trailing newline

Found 1 error.
[*] 1 fixable with the `--fix` option.
```

The following would help:
* Ruff version (`ruff --version`)
* How are you running Ruff? Editor / CLI?
* If from CLI, which command are you using to run?
* Ruff configuration which is being used here if any

**Comment by boolean-light** [comment]
I haven't changed the Ruff configurations.
I've directly cloned this Ruff repository, so if I put `cargo run -p ruff -- --version`, it outputs `ruff 0.3.2`.
I run Ruff with `cargo run -p ruff check src/W292.py --no-cache` and didn't get the warning.

Actually, I've tried the command and found that when I put the option of `--select=W292` it worked (gave me error), so it's probably a bug.
Sorry if this was just me confusing, but I'm not sure what's the exact rule which is enabled when I don't put anything in select option...?

**Comment by dhruvmanila** [comment]
> Actually, I've tried the command and found that when I put the option of `--select=W292` it worked (gave me error), so it's probably a bug.
> Sorry if this was just me confusing, but I'm not sure what's the exact rule which is enabled when I don't put anything in select option...?

No need to apologize. You can find the default rule set here: https://github.com/astral-sh/ruff#configuration

> ```toml
> [tool.ruff.lint]
> # Enable Pyflakes (`F`) and a subset of the pycodestyle (`E`)  codes by default.
> select = ["E4", "E7", "E9", "F"]
> ```

Closing this to keep the issue tracker actionable but feel free to ask any other questions that you might have.
